lib/ui/io.rb

```diff
@@ -7,7 +7,12 @@ def display_message(message)
     @output.print message
   end
 
-  def get_input
-    @input.gets.chomp
+  def get_input(message, callback)
+    input = @input.readline(message)
+    unless(input)
+      callback.call
+      exit
```

Comment:
As a rule, I avoid using `exit` outside of my main class. Putting it here can make it hard to test.

Reply:
Moved exit to one location in app.